src/gt4py/next/program_processors/runners/dace_fieldview/gtir_to_sdfg.py

```diff
@@ -399,9 +400,14 @@ def visit_Lambda(
         lambda_args_mapping = {str(p.id): arg for p, arg in zip(node.params, args, strict=True)}
 
         # inherit symbols from parent scope but eventually override with local symbols
-        lambda_symbols = self.global_symbols | {
-            pname: type_ for pname, (_, type_) in lambda_args_mapping.items()
-        }
+        referenced_symbols = set(
+            symbol_ref_utils.collect_symbol_refs(node.expr, self.global_symbols.keys())
+        )
+        lambda_symbols = {
+            pname: type_
+            for pname, type_ in self.global_symbols.items()
+            if pname in referenced_symbols
+        } | {pname: type_ for pname, (_, type_) in lambda_args_mapping.items()}
```

Comment:
This last line is almost lost and might be overlooked.
I (personally) would put it on its own line, i.e. 
```python
lambda_symbol |= {pname: type_ for pname, (_, type_) in lambda_args_mapping.items()}
```
or using `dict.update`.